ctlearn/data_loading.py

```diff
@@ -162,17 +185,25 @@ def __init__(self,
             self.map_fn_output_dtypes = data_dtypes + label_dtypes
                     
         elif self.example_type == 'array':
-            self.generator_output_dtypes = [np.dtype(np.int64), np.dtype(np.int64)] 
-           
-            data_dtypes = [np.dtype(np.float32), 
-                   np.dtype(np.int8),
-                   np.dtype(np.float32)
-                   ]
-
+            self.generator_output_dtypes = [np.dtype(np.int64),
+                    np.dtype(np.int64)] 
+            
+            tel_types = (['telescope'] if self.merge_tel_types 
+                    else self.selected_telescope_types)
+            data_output_names = []
+            data_dtypes = []
+            for tel_type in tel_types:
+                data_output_names.extend([tel_type + '_data',
+                    tel_type + '_triggers', tel_type + '_aux_inputs'])
+                data_dtypes.extend([np.dtype(np.float32), np.dtype(np.int8),
```

Comment:
Maybe we could pull these dtypes out to a config file or somewhere else, as they are not necessarily obvious or clearly indicated.

Reply:
I put these in dictionaries to make their meanings more obvious in 3028054. If there's a call for it I suppose we could pull them out to the config file.